smif/sos_model.py

```diff
@@ -146,22 +148,104 @@ def _run_static_sos_model(self):
         run_order = self._get_model_names_in_run_order()
         timestep = self.timesteps[0]
 
-        for model_name in run_order:
-            logging.debug("Running %s", model_name)
-            sector_model = self.model_list[model_name]
-            self._run_sector_model_timestep(sector_model, timestep)
+        for model_set in run_order:
+            self._run_model_set(model_set, timestep)
 
     def _run_sequential_sos_model(self):
         """Runs the system-of-system model sequentially
-
         """
         run_order = self._get_model_names_in_run_order()
         self.logger.info("Determined run order as %s", run_order)
         for timestep in self.timesteps:
-            for model_name in run_order:
-                logging.debug("Running %s for %d", model_name, timestep)
-                sector_model = self.model_list[model_name]
-                self._run_sector_model_timestep(sector_model, timestep)
+            for model_set in run_order:
```

Comment:
Should a model_set be a class with methods `run` and `guess_results` and property `converged`?